packages/metrics/src/Metrics.ts

```diff
@@ -917,6 +909,22 @@ class Metrics extends Utility implements MetricsInterface {
     this.envVarsService = new EnvironmentVariablesService();
   }
 
+  /**
+   * Set the function name for the cold start metric.
+   *
+   * @param functionName - The function name to be used for the cold start metric set in the constructor
+   */
+  protected setFunctionNameForColdStartMetric(functionName?: string): void {
+    const constructorFunctionName = functionName;
+    const envFunctionName = this.getEnvVarsService().getFunctionName();
+    for (const value of [constructorFunctionName, envFunctionName]) {
+      if (value && value.trim().length > 0) {
+        this.functionName = value;
+        break;
+      }
```

Comment:
Same case I mentioned before. If the condition is directly related to the value of the variable being present or not, why not if?